client/packages/programs/src/JsonForms/common/components/Select.tsx

```diff
@@ -168,22 +185,69 @@ const getHighlightParts = (
   );
 };
 
+/** Either the full list or the narrowed list from conditional option */
+
+const useFilteredItems = (
+  items: string[] | undefined,
+  current: string | undefined,
+  options: Options | undefined
+): [string[], string | undefined] => {
+  const { core } = useJsonForms();
+  const [error, setError] = useState<string | undefined>();
+  const [visibleItems, setVisibleItems] = useState(items ?? []);
+
+  const conditionField = extractProperty(
+    core?.data ?? {},
+    options?.fieldFilter?.field ?? ''
+  );
+  useEffect(() => {
+    if (!items || !options?.fieldFilter) {
+      return;
+    }
```

Comment:
Maybe it would be a bit clearer to escape early when there's no `conditionField`, as none of the mapping stuff applies:
```
if (!conditionField) {
  setVisibleItems(items)
  setError()
  return
}

```

Reply:
I can't return before a hook but I added the "unset" calls in the the useEffect hook

Reply:
Yeah, I meant return early *within* the useEffect hook:
```ts
useEffect(() => {
    if (!allItems || !options?.fieldFilter) {
      setVisibleItems(allItems ?? []);
      setError(undefined);
      return;
    }

    if (!conditionField) {
      setVisibleItems(items)
      setError(undefined)
      return
    }

   // The following only applies if we have a condition to deal with:
    const mapping = options.fieldFilter.mapping[conditionField] ?? [];
    const filtered = allItems.filter(item => mapping.includes(item));
    if (
      currentSelection !== undefined &&
      !filtered.includes(currentSelection)
    ) {
      setVisibleItems([currentSelection, ...filtered]);
      setError('Please select a valid option');
    } else {
      setVisibleItems(filtered);
      setError(undefined);
    }
```

Reply:
Ok I see, thanks I leave it though to reduce the number of setError, setVisibleItems...